bulkimport_scripts/add_users_for_bulk_import/index.js

```diff
@@ -0,0 +1,191 @@
+          describe: 'Path to the file storing users with invalid schema',
+          default: './usersHavingInvalidSchema.json',
+      })
+      .option('remaining-users-file', {
+          alias: 'r',
+          type: 'string',
+          describe: 'Path to the file storing remaining users',
+          default: './remainingUsers.json',
+      })
+      .argv;
+
+  return {
+      coreAPIUrl: argv['core-endpoint'],
+      inputFileName: argv['input-file'],
+      usersHavingInvalidSchemaFileName: argv['invalid-schema-file'],
+      remainingUsersFileName: argv['remaining-users-file'],
+  };
+}
+
+async function getUsersFromInputFile({ inputFileName }) {
+  try {
+    const inputFileDataString = await fs.readFile(inputFileName, 'utf8');
+    const inputFileData = JSON.parse(inputFileDataString);
+
+    if (!inputFileData.users || !Array.isArray(inputFileData.users) || inputFileData.users.length === 0) {
+      throw new Error('Expected users array in the input file.');
+    }
+
+    return inputFileData.users;
+  } catch (error) {
+    console.error('Error reading or parsing input file:', error.message);
+    process.exit(1);
+  }
+}
+
+async function deleteUsersHavingInvalidSchemaFileIfExists({ usersHavingInvalidSchemaFileName }) {
+  try {
+    await fs.rm(usersHavingInvalidSchemaFileName);
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      console.error(`Failed to delete ${usersHavingInvalidSchemaFileName}:`, error.message);
+    }
+  }
+}
+
+async function addInvalidSchemaUsersToFile({ errors, users, usersHavingInvalidSchemaFileName }) {
+  let parsedData = null;
+  try {
+    const existingData = await fs.readFile(usersHavingInvalidSchemaFileName, 'utf8');
+    parsedData = JSON.parse(existingData);
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      parsedData = { users: [] };
+    } else {
+      console.error(`Failed to read output file. Error: ${error.message}`);
+      throw error;
+    }
+  }
+
+  parsedData.users.push(...errors.map((err) => ({ user: users[err.index], errors: err.errors })));
+
+  await fs.writeFile(usersHavingInvalidSchemaFileName, JSON.stringify(parsedData, null, 2));
+
+  return users.filter((_, index) => !errors.some(err => err.index === index));
+}
+
+async function updateRemainingUsersFile({ users, index, remainingUsersFileName }) {
+  const remainingUsers = users.slice(index + 1);
+  await fs.writeFile(remainingUsersFileName, JSON.stringify({ users: remainingUsers }, null, 2));
+}
+
+async function removeRemainingUsersFile({ remainingUsersFileName }) {
+  try {
+    await fs.rm(remainingUsersFileName);
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      console.error(`Failed to delete ${remainingUsersFileName}:`, error.message);
+    }
+  }
+}
+
+async function main() {
+  const { coreAPIUrl, inputFileName, usersHavingInvalidSchemaFileName, remainingUsersFileName } = await parseInputArgs();
+
+  const users = await getUsersFromInputFile({ inputFileName });
```

Comment:
shouldnt this actually pick up from the remainingUsers file if that exists? Cause the input file would have users that have been successfully imported as well.

Reply:
Otherwise the dev would have to manually copy the contents of the remainingUsers file to their input file on each run, which can be something that they miss.